### PR TITLE
Suggest `_` in turbofish if param will be inferred from fn argument

### PR DESCRIPTION
**Pull request by estebank**

### CODE REVIEW

File: compiler/rustc_typeck/src/structured_errors/wrong_number_of_generic_args.rs [hunk 1]
```diff
@@ -292,12 +292,37 @@ impl<'a, 'tcx> WrongNumberOfGenericArgs<'a, 'tcx> {
...
+        let is_used_in_input = |def_id| match node {
+            Some(hir::Node::Item(hir::Item { kind: hir::ItemKind::Fn(fn_sig, _, _), .. }))
+            | Some(hir::Node::TraitItem(hir::TraitItem {
+                kind: hir::TraitItemKind::Fn(fn_sig, _),
+                ..
+            }))
+            | Some(hir::Node::ImplItem(hir::ImplItem {
+                kind: hir::ImplItemKind::Fn(fn_sig, _),
+                ..
+            })) => fn_sig.decl.inputs.iter().any(|ty| match ty.kind {
```
Comment (oli-obk) [inline compiler/rustc_typeck/src/structured_errors/wrong_number_of_generic_args.rs:305]: Someone was so kind and wrote a function that does this signature extraction: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/hir/map/fn.fn_sig.html

Comment (estebank) [reply compiler/rustc_typeck/src/structured_errors/wrong_number_of_generic_args.rs]: Changed.

**Comment by oli-obk** [comment]
r=me with CI passing

**Pull request merged**